mubench.pipeline/tasks/implementations/publish_metadata.py

```diff
@@ -54,7 +55,17 @@ def run(self, project: Project, misuse: Misuse):
 
     def __get_snippets(self, misuse, version):
         version_compile = version.get_compile(self.compiles_base_path)
-        return [snippet.__dict__ for snippet in misuse.get_snippets(version_compile.original_sources_path)]
+
+        snippets = []
+        for source_path in version_compile.original_sources_paths:
```

Comment:
It seems we have this logic (finding a file in one of the sources/classes directories) in multiple places. Can we centralize and reuse it?